Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/question/components/Draggable.kt

```diff
@@ -0,0 +1,187 @@
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.layout.boundsInWindow
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.layout.positionInWindow
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.center
+
+class DragTargetInfo<T> {
+    var isDragging: Boolean by mutableStateOf(false)
+    var hasDragEnd: Boolean by mutableStateOf(false)
+    var hasDragEndInAnyDropZone: Boolean by mutableStateOf(false)
+    var dragPosition by mutableStateOf(Offset.Zero)
+    var dragOffset by mutableStateOf(Offset.Zero)
+    var draggableComposable by mutableStateOf<(@Composable (Boolean) -> Unit)?>(null)
+    var dataToTransfer by mutableStateOf<T?>(null)
+}
+
+val LocalDragTargetInfo = compositionLocalOf { DragTargetInfo<Any?>() }
+
+@Composable
+fun Draggable(
+    modifier: Modifier = Modifier,
+    content: @Composable BoxScope.() -> Unit
+) {
+    val state = remember { DragTargetInfo<Any?>() }
+    CompositionLocalProvider(
+        LocalDragTargetInfo provides state
+    ) {
+        Box(modifier = modifier.fillMaxSize()) {
+            content()
+            if (state.isDragging) {
+                var targetSize by remember {
+                    mutableStateOf(IntSize.Zero)
+                }
+                var currentPosition by remember {
+                    mutableStateOf(Offset.Zero)
+                }
+                Box(
+                    modifier = Modifier
+                        .onGloballyPositioned {
+                            currentPosition = it.windowToLocal(state.dragPosition)
+                            targetSize = it.size
+                        }
+                        .graphicsLayer {
+                            val offset = (currentPosition + state.dragOffset)
+                            alpha = if (targetSize == IntSize.Zero) 0f else 1f
+                            translationX = offset.x.minus(targetSize.center.x)
+                            translationY = offset.y.minus(targetSize.center.y)
+                        }
+                ) {
+                    state.draggableComposable?.invoke(false)
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun <T> DragTarget(
+    onItemDropped: (T) -> Unit,
+    item: T,
+    modifier: Modifier = Modifier,
+    content: @Composable ((isDragging: Boolean) -> Unit)
+) {
+    var currentPosition by remember { mutableStateOf(Offset.Zero) }
+    val state = LocalDragTargetInfo.current
+
+    // Local dragging states
+    var isItemDragging by remember { mutableStateOf(false) }
+    var hasItemDraggingEnd by remember { mutableStateOf(false) }
+
+    // Check if any item dropped and this is the item
+    if (state.hasDragEndInAnyDropZone && hasItemDraggingEnd) {
+        hasItemDraggingEnd = false
+        state.hasDragEndInAnyDropZone = false
+        onItemDropped(item)
+    }
+
+    if (isItemDragging) {
+        state.dataToTransfer = item
+    }
```

Comment:
Same thing here - I dont believe we should be writing to state in the composition